googleapiclient/discovery.py

```diff
@@ -774,9 +774,21 @@ def method(self, **kwargs):
     if media_filename:
       # Ensure we end up with a valid MediaUpload object.
       if isinstance(media_filename, six.string_types):
-        (media_mime_type, encoding) = mimetypes.guess_type(media_filename)
+        # FIXME: get this from kwargs['media_mime_type'], rather than mucking with body
```

Comment:
@jonparrott That's what I was thinking. I discussed this internally on the customer issue. Are you proposing a warning if the `media_mime_type` is not explicitly defined? That might be annoying for existing users given that this is a new feature, and they've always relied on auto-detect before.

@nathanielmanistaatgoogle I use `FIXME` deliberately when I want to address something _prior_ to merge, whereas `TODO` is for something to be fixed at some point in the future (post-merge). In a large PR, I can `git grep FIXME` to make sure I didn't miss anything.